kafka/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KafkaSink.scala

```diff
@@ -40,7 +40,7 @@ class KafkaSink[F[_]: Sync](
     * @param events The list of events to send
     * @param key The partition key to use
     */
-  override def storeRawEvents(events: List[Array[Byte]], key: String): F[Unit] = Sync[F].delay {
+  override def storeRawEvents(events: List[Array[Byte]], key: String): F[Unit] = Sync[F].blocking {
```

Comment:
This line stood out to me as not ideal.  If we use `Sync[F].blocking` here then every single Event needs to be shifted to the blocking thread pool.  A task then runs on that blocking thread pool, which executes `kafkaProducer.send` (line 46).  After that is finished, processing shifts back onto the first thread pool again where we send the response back to the client.  That's a lot of thread shifting for a lot of events.

I understand that this is a workaround because `kafkaProducer.send` is blocking under some circumstances.  But I was really hoping we could find a way to make that non-blocking, instead of shifting blocking calls to a different thread pool.

Sorry for the negative opinion!!  If this is the only to fix it then of course I'm ok with that.

Reply:
I've got an alternative approach that uses fs2-kafka's Producer (not a full fs2 stream) that does a two-step operations on it that also seems to be working. I decided to put this one here, but I'll remove it and add another PR with just the fs2-kafka's producer.

Reply:
@istreeter I've reorganised it a bit, removed the `blocking` commit and organised everything by adding an additional fs2-kafka commit into the PR. I'm happy to split it up into separate PRs.